logical_type.go

```diff
@@ -304,6 +308,69 @@ func makeDecimalFixedCodec(st map[string]*Codec, enclosingNamespace string, sche
 	return c, nil
 }
 
+func makeValidatedStringCodec(st map[string]*Codec, enclosingNamespace string, schemaMap map[string]interface{}) (*Codec, error) {
+	patternFn, ok := schemaMap["pattern"]
+	if !ok {
+		return nil, errors.New("cannot create validated-string logical type without pattern")
+	}
+	if _, ok := schemaMap["name"]; !ok {
+		schemaMap["name"] = "string.validated-string"
+	}
+	c, err := registerNewCodec(st, schemaMap, enclosingNamespace)
+	if err != nil {
+		return nil, err
+	}
+
+	pattern := strings.TrimSpace(patternFn.(string))
+	if m[pattern] == nil {
+		m[pattern] = regexp.MustCompile(pattern)
```

Comment:
I recommend refactoring this line so that it does not panic when `pattern` does not compile properly as a legal regular expression.